app/jobs/application_job.rb

```diff
@@ -1,2 +1,6 @@
 class ApplicationJob < ActiveJob::Base
+  # https://api.rubyonrails.org/v5.2.1/classes/ActiveJob/Exceptions/ClassMethods.html
+  discard_on ActiveJob::DeserializationError do |_job, error|
```

Comment:
Isn't it suppressing the error? Should we try to fix the reason why deserialization error occurs? I guess this is to remove the errors like "Conversation with id not found" and "contact with id not found" errors in job.

Reply:
The error occurs because you are passing in an object as an argument and that object doesn't exist in the database.  [Either the object is deleted / its a race condition where the job is picked up before the item is committed to the database.]

After looking into logs the primary cases I have seen are 
- Contact id not found ( Result of merge action where original contact was deleted ) 
- Contact Avatar job ( same ) 
- Conversation Id ( conversation wasn't committed cause of some error. but event dispatch job was called on update ) 

There isn't much we can do to recover/reprocess the job in case of these deserialization errors.